src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -1294,36 +1294,59 @@ public LazyConditionalChoiceExpression(Lazy<IOperation> condition, Lazy<IOperati
+        where TConversion : struct, IConversion
     {
-        protected BaseConversionExpression(ConversionKind conversionKind, bool isExplicit, bool usesOperatorMethod, IMethodSymbol operatorMethod, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+        protected BaseConversionExpression(TConversion conversion, bool isExplicitInCode, bool throwsExceptionOnFailure, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
                     base(OperationKind.ConversionExpression, syntax, type, constantValue)
         {
-            ConversionKind = conversionKind;
-            IsExplicit = isExplicit;
-            UsesOperatorMethod = usesOperatorMethod;
-            OperatorMethod = operatorMethod;
+            ConversionInternal = conversion;
+            IsExplicitInCode = isExplicitInCode;
+            ThrowsExceptionOnFailure = throwsExceptionOnFailure;
         }
         /// <summary>
+        /// Internal direct conversion access. Used for the non-boxing language specific conversion methods.
+        /// </summary>
+        internal TConversion ConversionInternal { get; }
+        /// <summary>
         /// Value to be converted.
         /// </summary>
         public abstract IOperation Operand { get; }
+#pragma warning disable CS1574 // XML comment has cref attribute that could not be resolved
+        // These crefs come from conversions defined in the C# and VB specific projects
         /// <summary>
-        /// Kind of conversion.
+        /// Gets the underlying conversion. This will be either <see cref="Microsoft.CodeAnalysis.CSharp.Conversion"/>
+        /// or <see cref="Microsoft.CodeAnalysis.VisualBasic.Conversion"/>.
         /// </summary>
-        public ConversionKind ConversionKind { get; }
+        /// <remarks>
+        /// This is a boxing operation: if you need conversion information that is language specific, use either
+        /// <see cref="Microsoft.CodeAnalysis.CSharp.IConversionExpressionExtensions.GetCSharpConversion(IConversionExpression)"/> or
+        /// <see cref="Microsoft.CodeAnalysis.VisualBasic.GetVisualBasicConversion(IConversionExpression)"/>, which do not allocate memory.
+        /// </remarks>
+        public IConversion Conversion => ConversionInternal;
+#pragma warning restore CS1574 // XML comment has cref attribute that could not be resolved
         /// <summary>
         /// True if and only if the conversion is indicated explicity by a cast operation in the source code.
         /// </summary>
-        public bool IsExplicit { get; }
+        public bool IsExplicitInCode { get; }
+        /// <summary>
+        /// True if the conversion will fail with an exception at runtime if the cast fails. This is false for C#'s <code>as</code>
+        /// operator and for VB's <code>TryCast</code> operator.
+        /// </summary>
```

Comment:
Perhaps we need a different wording here, maybe will throw a <see cref="InvalidCastException" /> at runtime? Checked vs unchecked is tracked by https://github.com/dotnet/roslyn/issues/18055, and I'm going to work a separate `IsChecked` boolean into this PR.